lib/vsc/filesystem/oceanstor.py

```diff
@@ -1810,3 +1810,141 @@ def _get_grace_expiration(grace, usage, soft, hard):
             expired = (False, None)
 
         return expired
+
+    def list_snapshots(self, filesystem, fileset=None):
+        """
+        List the snapshots in the given filesystem or dtree fileset
+
+        @type filesystem: name of the filesystem
+        @type fileset: name of the dtree fileset
+        """
+
+        fs = self.get_filesystem_info(filesystem)
+        filter_json = {"file_system_id": int(fs["id"])}
+
+        if fileset is not None:
+            dtree = self.get_fileset_info(filesystem, fileset)
+            if dtree is None:
+                err_msg = "Snapshot query failed: fileset '%s' not found in filesystem '%s'" % (filesystem, fileset)
+                self.log.raiseException(err_msg, OceanStorOperationError)
+            filter_json.update({"dtree_id": dtree["id"]})
```

Comment:
any particular reason you're not just adding it in the dict directly? `filter_json['dtree_od'] = dtree['id']`?

Reply:
nothing in particular, changed in https://github.com/vub-hpc/vsc-filesystem-oceanstor/pull/24/commits/71038c50ba8a005f08da15dcb614474a54f1a9a4